include/mbedtls/pk.h

```diff
@@ -172,10 +252,17 @@ typedef size_t (*mbedtls_pk_rsa_alt_key_len_func)( void *ctx );
 #endif /* MBEDTLS_PK_RSA_ALT_SUPPORT */
 
 /**
- * \brief           Return information associated with the given PK type
+ * \brief           Return default information associated with the given PK type
  *
  * \param pk_type   PK type to search for.
  *
+ * \note            Different PK objects with the same type may have different
```

Comment:
I'm having second thoughts about this note. I think I wrote it back when I thought that `mbedtls_pk_get_type` would return e.g. `MBEDTLS_PK_RSA` for an opaque RSA key, which turned out not to work out with how `mbedtls_pk_context` is laid out. In fact, it remains the case that all objects whose type is `MBEDTLS_PK_RSA` objects are represented as a `mbedtls_rsa_context`.

Proposal: replace the note by

> This function returns NULL if `pk_type` indicates an opaque key, since the type does not provide enough information to build an opaque key.

@hanno-arm Thoughts?

This function also returns NULL for `MBEDTLS_PK_RSAPSS`, @mpg do you know why?

Reply:
Done